### PR TITLE
Random sample on stream

### PR DESCRIPTION
**Pull request by vlejd**
Related to #308.

Fix comments in  #1422.

### CODE REVIEW

File: gensim/corpora/textcorpus.py [hunk 1]
```diff
@@ -130,11 +130,14 @@ def sample_texts(self, n, seed=None, length=None):
...
-            raise ValueError("n is larger than length of corpus.")
+
+            raise ValueError("n {0:d} is larger than length of corpus {1:d}.".format(n, length))
         if not 0 <= n:
-            raise ValueError("Negative sample size.")
+            raise ValueError("Negative sample size n {0:d}.".format(n))
 
+        seen_documents = 0
         for i, sample in enumerate(self.get_texts()):
+            seen_documents = i + 1
```
Comment (piskvorky) [inline gensim/corpora/textcorpus.py]: What is the purpose of introducing this new variable?

Comment (piskvorky) [inline gensim/corpora/textcorpus.py]: Oh I think I see -- you can directly re-use the counter `i` from enumerate (just give it a better name).

Comment (vlejd) [inline gensim/corpora/textcorpus.py]: It may be only my personal opinion, but I do not like this type of "variable leaking" out of the for loop scope.

Comment (piskvorky) [inline gensim/corpora/textcorpus.py]: OK, sure. Then I suggest `seen_documents += 1` at least.

**Comment by menshikh-iv** [comment]
Ping @vlejd, when you finish your PR?

**Comment by menshikh-iv** [comment]
Thank you @vlejd

File: gensim/corpora/textcorpus.py [hunk 2]
```diff
@@ -305,7 +302,7 @@ def sample_texts(self, n, seed=None, length=None):
             # This means that length was set to be greater than number of items in corpus
             # and we were not able to sample enough documents before the stream ended.
             raise ValueError(
-                "length {0:d} greater than number of documents in corpus {1:d}".format(length, seen_documents))
+                "length {0:d} greater than number of documents in corpus {1:d}".format(length, i + 1))
```
Comment (piskvorky) [inline gensim/corpora/textcorpus.py]: This will fail with an exception if there are no texts (`i` will be uninitialized).

Comment (menshikh-iv) [inline gensim/corpora/textcorpus.py]: `i` undefined if only `self.getstream()` is empty

**Pull request merged**